ui/src/routes/devices.$id.settings.appearance.tsx

```diff
@@ -48,6 +77,24 @@ export default function SettingsAppearanceRoute() {
           }}
         />
       </SettingsItem>
+      <SettingsItem title="Device Name" description="Set your device name">
+        <InputFieldWithLabel
+          required
+          label=""
+          placeholder="Enter Device Name"
+          description={`DNS: ${nameConfigSettings.dns}`}
+          defaultValue={nameConfigSettings.name}
+          onChange={e => handleNameChange(e.target.value)}
+        />
+      </SettingsItem>
```

Comment:
Add ability to edit DNS with something like this?
```tsx
 <SettingsItem title="DNS Name" description="Set the DNS name for the device">
    <InputFieldWithLabel
      required
      label=""
      placeholder="Enter DNS Name"
      defaultValue={nameConfigSettings.dns}
      onChange={e => handleDNSChange(e.target.value)}
    />
  </SettingsItem>
```
If we leave off the `required` then they could leave it blank to allow the API to compute the default suffix of `'.local'` or we could just do that in the `setDNS` state manager.

```tsx
const [dns, setDNS] = useState("");

const handleDNSChange = (value: string) => {
  // to do client-side defaulting uncomment
  // if (value.len() == 0) {
  //   value = name + ".local"
  // }
  setDNS(value);
};
```